consumer.go

```diff
@@ -669,6 +669,11 @@ func (r *Consumer) startStopContinueBackoff(conn *Conn, success bool) {
 			var i int
 			conns := r.conns()
 			if len(conns) == 0 {
+				r.backoffMtx.Lock()
+				if r.backoffCounter > 0 {
+					r.backoffCounter--
+				}
+				r.backoffMtx.Unlock()
```

Comment:
If we treat backoff as backing off from downstream systems, then i agree. It should persist beyond upstream nsqd reconnects

Since this code block is in a `time.After` we should be able to refactor to just re-set that function recursively (and have it end on exit flags appropriately).